iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -47,7 +47,7 @@
 #define BUILD_CONFIG_USERNAME               24
 #define SAS_TOKEN_DEFAULT_LEN               10
 #define RESEND_TIMEOUT_VALUE_MIN            1*60
-#define MAX_SEND_RECOUNT_LIMIT              2
+#define MSG_TIMEOUT_VALUE_MIN               2*60
```

Comment:
The current implementation of 2 retries counts the first send as a retry, so it's actually only 1 retry from what I've seen (let me know if this is incorrect, I may have been testing this in a weird way!). I set it to 2 minutes so that we have the most lee-way with any delays like you mentioned without the possibility of sending the message 3 times (I check below for >= 2 minutes so in the instance where everything is optimally speedy and the third message tries to send at exactly 2 minutes, the message will still get expired).

Reply:
Yup, you're totally right I'd forgot that.  The code here with the 2 minutes is fine indeed.

Which is another good reason I guess to get rid of this retryCount since it wasn't mapping up to how one would naively think it'd work.